hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/metrics/MetricNames.java

```diff
@@ -95,26 +92,59 @@
      * of a particular incoming {@link Edge} of a certain {@link Vertex}.
-     * The {@link Edge} and the {@link Vertex}, together with the concrete
-     * {@link Processor} running them can be identified based on the
-     * {@link MetricTags#ORDINAL}, {@link MetricTags#VERTEX} &
-     * {@link MetricTags#PROCESSOR} tags of the metric.
+     * The {@link Vertex} and the {@link Processor} can be identified
+     * based on the {@link MetricTags#VERTEX} & {@link MetricTags#PROCESSOR}
+     * tags of the metric.
      */
     public static final String COALESCED_WM = "coalescedWm";
 
     /**
-     * Tracks the last watermark emitted by a particular {@link Processor},
-     * which can be identified based on the {@link MetricTags#PROCESSOR} tag.
+     * Tracks the last watermark emitted by a particular {@link Processor}
+     * of a particular {@link Vertex}. The {@link Vertex} and the
+     * {@link Processor} can be identified based on the {@link MetricTags#VERTEX}
+     * & {@link MetricTags#PROCESSOR} tags of the metric.
      */
     public static final String LAST_FORWARDED_WM = "lastForwardedWm";
 
     /**
      * Tracks the difference between the last emitted watermark and the
-     * system time of a particular {@link Processor}. The {@link Processor}
-     * can be identified based on the {@link MetricTags#PROCESSOR} tag.
+     * system time of a particular {@link Processor} of a particular
+     * {@link Vertex}. The {@link Vertex} and the {@link Processor} can be
+     * identified based on the {@link MetricTags#VERTEX} &
+     * {@link MetricTags#PROCESSOR} tags of the metric.
      */
     public static final String LAST_FORWARDED_WM_LATENCY = "lastForwardedWmLatency";
 
+    /**
+     * Tracks the total number of bytes written in the last snapshot
+     * by a particular {@link Vertex}. The name of the vertex can be found
+     * in the {@link MetricTags#VERTEX} tag of the metric.
+     */
+    public static final String SNAPSHOT_BYTES = "snapshotBytes";
+
+    /**
+     * Tracks the total number of keys written in the last snapshot
+     * by a particular {@link Vertex}. The name of the vertex can be found
+     * in the {@link MetricTags#VERTEX} tag of the metric.
+     */
+    public static final String SNAPSHOT_KEYS = "snapshotKeys";
+
+    /**
+     * Tracks the start time of a given execution of a specific job.
+     * The execution and the job can be identified based on the
+     * {@link MetricTags#EXECUTION} & {@link MetricTags#JOB} tags of
+     * the metric.
+     */
+    public static final String EXECUTION_START_TIME = "executionStartTime";
+
+    /**
+     * Tracks the completion time of a given execution of a specific job.
+     * The execution and the job can be identified based on the
+     * {@link MetricTags#EXECUTION} & {@link MetricTags#JOB} tags of
+     * the metric.
+     */
+    public static final String EXECUTION_COMPLETION_TIME = "executionCompletionTime";
```

Comment:
Isn't the execution gone as soon as the job is completed? One can observe a non-zero value in a very short window.
If not, we should document that it's 0 when the job is running.

Reply:
it should be in the completed jobs metrics, I think?

Reply:
I realized it might be useful to have executionCompletionTime if one stores metrics after job completion. I added a test for it and fixed it...

Reply:
Which then brakes other things...

Reply:
Pardon my momentary frustration... Well it's not momentary, stopping it from coming out is.